### PR TITLE
WIP: Add Shellcheck to Pre-commit and Batch fix issues

### PR DESCRIPTION
**Pull request by msnwatson**
## What was the problem?

We had many warnings and improvements for implementing best practices in our shell scripts.

## How does this fix it?[^1]

In order to ensure strong code quality going forward, I added the shellcheck tool to run as a pre-commit hook. I also ran pre-commit on all files in the repo (except those explicitly excluded) to get us to a clean state with our pre-commit tool.

## How to test this PR
IN PROGRESS

[^1]: [Pull-Requests guidelines](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Pull-Requests). If PR is significant, update [Current Software State](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Current-Software-State) wiki page.
[^secrel]: To check if a PR will succeed in the SecRel workflow, [test PRs in the SecRel pipeline](https://github.com/department-of-veterans-affairs/abd-vro-internal/wiki/Secure-Release-process#to-test-prs-in-the-secrel-pipeline).

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -51,6 +52,11 @@ repos:
     hooks:
       - id: gitleaks
 
+  - repo: https://github.com/jumanjihouse/pre-commit-hooks
+    rev: 3.0.0
+    hooks:
+      - id: shellcheck
+
```
Comment (yoomlam) [inline .pre-commit-config.yaml:59]: Crux of PR

File: .github/runner/set-k8s-secrets.sh [hunk 2]
```diff
@@ -53,14 +53,14 @@ $2
 }
 
 splitSecretData(){
-  echo "$1" | jq -r 'to_entries | .[] | "\(.key) \(.value)"' | while read K V; do
+  echo "$1" | jq -r 'to_entries | .[] | "\(.key) \(.value)"' | while read -r K V; do
```
Comment (yoomlam) [inline .github/runner/set-k8s-secrets.sh:56]: What's the reason for using `-r`? [Restricted mode](https://www.gnu.org/software/bash/manual/html_node/The-Restricted-Shell.html) prevents "changing directories" and "redirecting output", among other things. Using `read -r` everywhere may make it hard to diagnose why some basic commands don't work.

File: .github/runner/set-k8s-secrets.sh [hunk 3]
```diff
@@ -2,7 +2,7 @@
 # This script runs in the gh-runner container to set K8s secrets needed by VRO containers.
 
 NAMESPACE=$(cat /run/secrets/kubernetes.io/serviceaccount/namespace)
-: ${TARGET_ENV:=${NAMESPACE#va-abd-rrd-}}
+: "${TARGET_ENV:=${NAMESPACE#va-abd-rrd-}}"
```
Comment (yoomlam) [inline .github/runner/set-k8s-secrets.sh:5]: The quotes provide no added benefit for the [colon operator](https://stackoverflow.com/questions/7444504/explanation-of-colon-operator-in-foo-value). e.g., `: asdf asdf asdf` and `: "asdf asdf asdf"` are both safe commands. The extra quotes just adds noise.

File: .github/runner/set-k8s-secrets.sh [hunk 4]
```diff
@@ -108,7 +108,7 @@ VRO_SECRETS_NAMES="VRO_SECRETS_API VRO_SECRETS_SLACK VRO_SECRETS_MAS VRO_SECRETS
 # and evaluated by set-env-secrets.src in each VRO container.
 # Advantage: New environment variables can be added to these secrets without modifying
 # Helm configurations -- simply add them to Vault.
-SECRET_DATA=$(collectSecretExportCmds $VRO_SECRETS_NAMES)
+SECRET_DATA=$(collectSecretExportCmds "$VRO_SECRETS_NAMES")
```
Comment (yoomlam) [inline .github/runner/set-k8s-secrets.sh:111]: Undo this change so that the values in `VRO_SECRETS_NAMES` are treated as separate arguments. 
```suggestion
SECRET_DATA=$(collectSecretExportCmds $VRO_SECRETS_NAMES)
```

Adding quotes treats it as a single argument:
```sh
 > VRO_SECRETS_NAMES="VRO_SECRETS_API VRO_SECRETS_SLACK"
 > hello(){ echo "$1, $2"; }
 > hello "$VRO_SECRETS_NAMES"
VRO_SECRETS_API VRO_SECRETS_SLACK,

 > hello $VRO_SECRETS_NAMES
VRO_SECRETS_API, VRO_SECRETS_SLACK
```

File: app/src/docker/entrypoint.sh [hunk 5]
```diff
@@ -3,6 +3,6 @@
 
 eval exec java -jar \
   -Djava.security.egd=file:/dev/./urandom \
-  $JAVA_PROFILE \
-  $JAVA_OPTS \
+  "$JAVA_PROFILE" \
+  "$JAVA_OPTS" \
```
Comment (yoomlam) [inline app/src/docker/entrypoint.sh:7]: Undo this change b/c `$JAVA_OPTS` may have space delimited arguments intended to be treated as separate arguments. Adding quotes causes the entire `$JAVA_OPTS` value to be treated as a single argument. The `eval exec` may help prevent that but testing is required.

**Review by yoomlam — CHANGES_REQUESTED** [review]

File: gradle-plugins/vis-network/dependency-graph.sh [hunk 6]
```diff
@@ -36,10 +36,12 @@ pluginUsageEdges(){
...
+export PLUGIN_DEPS
 cd - || exit 21
 cd ../.. || exit 22
-export PLUGIN_USES=$(pluginUsageEdges)
+PLUGIN_USES=$(pluginUsageEdges)
+export PLUGIN_USES
 cd - || exit 23
 
-cat dependency-graph-template.html | envsubst > dependencies.html
+cmd dependency-graph-template.html | envsubst > dependencies.html
```
Comment (yoomlam) [inline gradle-plugins/vis-network/dependency-graph.sh:47]: This change doesn't look correct. 
```sh
> cmd dependency-graph-template.html
bash: cmd: command not found
```

File: helm/deploy.sh
Comment (yoomlam) [inline helm/deploy.sh:1]: I don't agree with any change in this file.

File: scripts/generate_sa_kubeconfig.sh
Comment (yoomlam) [inline scripts/generate_sa_kubeconfig.sh:1]: I haven't used this file.

File: scripts/image-version.sh [hunk 7]
```diff
@@ -20,28 +24,36 @@ unpinImageVersion(){
   sed "/^${IMG_VAR}=/d" scripts/image_versions.src
 }
 pinnedImages(){
-  for PREFIX in ${VAR_PREFIXES_ARR[@]}; do
+  for PREFIX in "${VAR_PREFIXES_ARR[@]}"; do
```
Comment (yoomlam) [inline scripts/image-version.sh:27]: Undo this change -- `${VAR_PREFIXES_ARR[@]}` should be treated as separate arguments.

File: scripts/image-version.sh [hunk 8]
```diff
@@ -4,9 +4,13 @@ source scripts/image_vars.src
 
 pinImageVersions(){
   echo "# $(date)"
-  for PREFIX in ${VAR_PREFIXES_ARR[@]}; do
-    local IMG_VER=$(getVarValue "${PREFIX}" _VER)
-    local IMG_VAR="${PREFIX}_VER"
+  for PREFIX in "${VAR_PREFIXES_ARR[@]}"; do
```
Comment (yoomlam) [inline scripts/image-version.sh:7]: Undo this change -- `${VAR_PREFIXES_ARR[@]}` should be treated as separate arguments.

File: scripts/set-secret-ghcr.sh [hunk 9]
```diff
@@ -19,7 +19,7 @@ case "${TARGET_ENV}" in
 esac
 
 SECRET_NAME="devops-ghcr"
-kubectl -n "va-abd-rrd-$TARGET_ENV" $KUBE_ARGS create secret docker-registry $SECRET_NAME \
+kubectl -n "va-abd-rrd-$TARGET_ENV" "$KUBE_ARGS" create secret docker-registry $SECRET_NAME \
```
Comment (yoomlam) [inline scripts/set-secret-ghcr.sh:22]: Undo this change -- `$KUBE_ARGS` should be treated as separate arguments.

File: scripts/setenv.sh [hunk 10]
```diff
@@ -2,7 +2,7 @@
 # This script is for local development or automated end2end testing.
 # No production secret values are in this file.
 
-if [[ $0 == $BASH_SOURCE ]]; then
+if [[ $0 == "${BASH_SOURCE[0]}" ]]; then
```
Comment (yoomlam) [inline scripts/setenv.sh:5]: I'm not familiar with this. Be sure to test this change.

**Review by yoomlam — CHANGES_REQUESTED** [review]

**Comment by msnwatson** [comment]
Moving back to draft state since original understanding of changes was not complete and more testing required

**Comment by yoomlam** [comment]
> Moving back to draft state since original understanding of changes was not complete and more testing required

Can bits of this PR be incrementally merged to our repo, after testing?

**Comment by lukey-luke** [comment]
changes to domain-cc look good

**Comment by yoomlam** [comment]
@msnwatson Still want to keep this open? There's a lot of conflicting files.
